votor/src/certificate_pool_service/stats.rs

```diff
@@ -0,0 +1,88 @@
+use {
+    solana_metrics::datapoint_info,
+    std::time::{Duration, Instant},
+};
+
+const STATS_REPORT_INTERVAL: Duration = Duration::from_secs(10);
+
+#[derive(Debug)]
+pub(crate) struct CertificatePoolServiceStats {
+    pub(crate) add_message_failed: u32,
+    pub(crate) certificates_sent: u16,
+    pub(crate) certificates_dropped: u16,
+    pub(crate) new_finalized_slot: u16,
+    pub(crate) new_parent_ready: u16,
+    pub(crate) new_parent_ready_missed_window: u16,
+    pub(crate) new_root: u16,
+    pub(crate) received_votes: u32,
+    pub(crate) received_certificates: u32,
```

Comment:
we should also add some stats for everytime we insert a new certificate (to cover both receiving and constructing) for each type by plugging this into `certificate_pool.rs`.

Reply:
That's a good point, but I think this datapoint_info! line already has quite some data. We should probably give certificate_pool its own stats and datapoint_info! lines, so we can count cert construction and cert construction errors as well. That should be another PR though.